www/js/survey/multilabel/multi-label-ui.js

```diff
@@ -236,29 +201,5 @@ angular.module('emission.survey.multilabel.buttons',
     trip.verifiability = someYellow ? "can-verify" : (allGreen ? "already-verified" : "cannot-verify");
   }
 
-  /*
-   * Embody the logic for delayed update:
-   * the recompute logic already keeps trips that are waitingForModification
-   * even if they would be filtered otherwise.
-   * so here:
-   * - set the trip as waiting for potential modifications
-   * - create a one minute timeout that will remove the wait and recompute
-   * - clear the existing timeout (if any)
-   */
-  mls.updateVisibilityAfterDelay = function(trip) {
-    // We have just edited this trip, and are now waiting to see if the user
-    // is going to modify it further
-    trip.waitingForMod = true;
```

Comment:
Yes. Before I removed this, it wasn't disappearing at all.
I'm not sure which is worse - labeled trips stick around in the "To Label" until you refresh or relaunch -- or the previous behavior where they stick around for some period of time and then disappear, seemingly at random.

I think ideally there should be an indication that the trip was labeled and then an animation. Maybe it can start to fade away, or be able to be swiped away. That is not a high priority at the moment though

Reply:
I am going to push back on this. We originally didn't have the delay, but added it on user request because otherwise, there was no easy way to correct mistakes that you made while labeling.

So the flow would be:
- I have a trip, I want to label "E-Car, Shared Ride" and "Shopping"
- I select "E-Car, Shared Ride"
- Due to an inadvertent fat finger press, I select "Meal" (the option after "Shopping")
- The trip disappears - aaaaaa!
- Now I have to go back to "All Trips" and scroll until I find the trip (assuming that I still remember which one it was) and then label it

This is very bad UX which is why we added the delay in the first place.

The `waitingForMod` code is still here. I suspect that the reason that it was not disappearing was that you changed this to use an angular broadcast to trigger the recompute and are not listening to it in angular any more.

You might want to call the equivalent of `repopulateTimelineEntry` from this function to indicate that the trip was actually modified.

Reply:
I understand why it's necessary to have a chance to correct errors.
I do want to find a solution where there's a period of time to revise the labels before the trip disappears, but when it does disappear, it's not surprising and doesn't feel like an accident.

I can prioritize reimplementing `waitingForMod` with your suggestion

Reply:
Also, my apologies for not noticing this earlier, but it looks like you also deleted `mls.updateTripProperties = function(trip) {`

That is definitely necessary - it should be called when we store the label to get the yellow trips to re-populate, enable verification etc

Here's where it is called before the rewrite:
https://github.com/e-mission/e-mission-phone/blob/e91c8d05a5e84cd1a538a5e76dbb5515fba748cc/www/js/survey/multilabel/multi-label-ui.js#L230

Here's where I think it should be called now:
https://github.com/e-mission/e-mission-phone/blob/b0d28d5cb5d8c62fd0d0b34be5d3149e99d46f88/www/js/survey/multilabel/MultiLabelButtonGroup.tsx#L88

`updateTripProperties` should be called while storing, it updates the trip properties correctly and then calls `repopulateTimelineEntry`

Reply:
`repopulateTimelineEntry()`, which is in `LabelTab`, does all this now.

React doesn't like when you mutate state, so we can't just change trip properties in-place the way `updateTripProperties` did. We have to set the state at its source (with `setTimelineMap()` in LabelTab, which `repopulateTimelineEntry()` calls). Otherwise React has no idea anything changed.

Reply:
Argh! I have to get used to thinking in "react-style" instead of "angular-style". Doesn't really help that I'm not actually writing any of this code. Once all the interns get on board, I want to make some UI changes too just to rewire by brain a little bit if possible

Reply:
still, TODO reinstate `waitingForMod` (or similar mechanism)